Result<Infallible, std::io::error::Error>
